core/src/org/otw/open/engine/impl/DragAndDropActorEngine.scala

```diff
@@ -0,0 +1,118 @@
+package org.otw.open.engine.impl
+
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.scenes.scene2d.utils.DragAndDrop
+import com.badlogic.gdx.scenes.scene2d.utils.DragAndDrop.{Payload, Source, Target}
+import com.badlogic.gdx.scenes.scene2d.{InputEvent, Stage}
+import com.badlogic.gdx.utils.viewport.ScreenViewport
+import org.otw.open.controllers.{CauseAndEffectFinishedSuccessfully, CauseAndEffectFinishedUnsuccessfully, ScreenController}
+import org.otw.open.dto.Drawing
+import org.otw.open.engine.Engine
+import org.otw.open.engine.actor.DragAndDropActor
+import org.otw.open.engine.util.SoundEffects
+
+/**
+  * Created by smirakovska on 2/9/2016.
+  *
+  * @param theme             - name of theme
+  * @param imgBackgroundPath - path to background image file
+  */
+class DragAndDropActorEngine(val theme: String, val imgBackgroundPath: String) extends Engine {
+
+  /**
+    * Stage object
+    */
+  var stage: Stage = _
+
+  /**
+    * Mutable instance of DragAndDropActor
+    */
+  private var actor = new DragAndDropActor(theme, imgBackgroundPath, 0, 320)
+
+  val dragAndDrop = new DragAndDrop()
+
+  /**
+    * Sound instance for cause and effect game
+    */
+  private val sound: SoundEffects = new SoundEffects("guidanceForCauseAndEffect.wav")
+
+  /**
+    * Number of failed attempts
+    */
+  private var failedAttempts = 0
+
+  /**
+    * Number of maximum allowed failed attempts
+    */
+  private val maxFailedAttempts = 3
+
+  stage = new Stage(new ScreenViewport())
```

Comment:
Is there no parameter constructor for Stage? This may cause problems with the resizing.